crates/proto/src/rr/rdata/txt.rs

```diff
@@ -156,6 +162,30 @@ mod tests {
         assert_eq!(rdata, read_rdata);
     }
 
+    /// Scenario
+    ///
+    /// Give invalid utf8 and valid ut8.
+    /// Shall return incorrect utf8 char for invalid utf8 known as:
+    /// `U+FFFD REPLACEMENT CHARACTER` looks `�`.
+    /// And correct utf8 output untouched.
+    #[test]
+    fn fmt_invalid_and_valid_utf8() {
+        // "Invalid utf8 <bytes garbage>."
+        let first_bytes = b"Invalid utf8 <\xF0\x90\x80>.";
+        // "Valid utf8 <🤣>"
+        let second_bytes = b" Valid utf8 <\xF0\x9F\xA4\xA3>";
+        let rdata: Vec<&[u8]> = vec![first_bytes, second_bytes];
```

Comment:
Style nit: I think we can do without the type annotation here? If so, that would be preferable.

Reply:
I has the same nit but it turns out that if not annotated I got this:

```rust
---- src/rr/rdata/txt.rs - rr::rdata::txt::TXT::fmt (line 135) stdout ----
error[E0308]: mismatched types
 --> src/rr/rdata/txt.rs:139:31
  |
7 | let rdata = vec![first_bytes, second_bytes];
  |                               ^^^^^^^^^^^^ expected an array with a fixed size of 19 elements, found one with 18 elements

error[E0308]: mismatched types
  --> src/rr/rdata/txt.rs:140:27
   |
8  | let txt = TXT::from_bytes(rdata);
   |           --------------- ^^^^^ expected slice `[u8]`, found array `[u8; 19]`
   |           |
   |           arguments to this function are incorrect
   |
   = note: expected struct `Vec<&[u8], std::alloc::Global>`
              found struct `Vec<&[u8; 19], _>`
```